src/main/scala/is/hail/io/bgen/BgenLoader.scala

```diff
@@ -76,13 +76,21 @@ object BgenLoader {
       (decoder.getKey, (decoder.getAnnotation, decoder.getValue))
     })).toOrderedRDD[Locus](fastKeys)
 
+    val noMulti = rdd.forall(_._1.nAlleles == 2)
```

Comment:
You should use fastKeys, not rdd.  You're just checking the keys.  Also, you probably want to persist the fastKeys like LoadVCF since they will be scanned twice in general (once to see if the keys are sorted, once to check for multi-allelics.)